manifests/qualys.pp

```diff
@@ -1,8 +1,165 @@
-# @summary A short summary of the purpose of this class
+# @summary Configure host to be scanned by qualys
 #
-# A description of what this class does
+# See https://wiki.ncsa.illinois.edu/display/SecOps/Qualys+Authenticated+Scanning+Host+setup
+#
+# @param enabled
+#   Boolean to define if authenticated qualys scan enabled
+#
+# @param gid
+#   String of the GID of the local qualys user
+#
+# @param group
+#   String of the group name of the local qualys user
+#
+# @param homedir
+#   String of the home directory path of the local qualys user
+#
+# @param ip
+#   String of the IP address that the qualys user can login from
+#
+# @param ssh_authorized_key
+#   String of the public ssh authorized key that the qualys user uses for authentication
+#
+# @param ssh_authorized_key_type
+#   String of the key type used for the qualys user's authentication
+#
+# @param uid
+#   String of the UID of the local qualys user
+#
+# @param user
+#   String of the username of the local qualys user
+#
+# @param user_comment
+#   String of the comment in passwd file of the local qualys user
 #
 # @example
 #   include profile_audit::qualys
-class profile_audit::qualys {
+#
+class profile_audit::qualys (
+  Boolean            $enabled,
+  String             $gid,
+  String             $group,
+  String             $homedir,
+  String             $ip,
+  Optional[ String ] $ssh_authorized_key,
+  String             $ssh_authorized_key_type,
+  String             $uid,
+  String             $user,
+  String             $user_comment,
+) {
+
+  # ONLY SETUP If enabled AND A ssh_authorized_key IS PROVIDED FOR QUALYS USER
+  if ( $enabled and $ssh_authorized_key )
```

Comment:
Should puppet "warn" if enabled is true but ssh_authorized_key is not provided? A warning will give feedback to the admin running puppet that something doesn't make sense, but it won't cause compilation to fail nor stop the puppet run.

Reply:
That is a good idea. Are you suggesting something like the following?
```
if ( $enabled and ! $ssh_authorized_key )
{
  notify {"Qualys is enabled, but no ssh authorized key has been supplied for the qualys user. A ssh authorized key must be supplied if qualys is enabled.":
    withpath => true,
  }
}
```

Reply:
That looks good. That and the original check should be combined into the same if/elseif control block.